File: Soe.Common/DTO/ActorSearchPersonDTO.cs
using SoftOne.Soe.Common.Util;
using System;


namespace SoftOne.Soe.Common.DTO
{
    public class ActorSearchPersonDTO
    {
        public int RecordId { get; set; }

        public string Number { get; set; }
        public string Name { get; set; }
        public string OrgNr { get; set; }
        public bool IsPrivatePerson { get; set; }
        public SoeEntityType EntityType { get; set; }
        public string EntityTypeName { get; set; }
        public bool HasConsent { get; set; }
        public string HasConsentString { get; set; }
        public DateTime? ConsentDate { get; set; }
    }
}

File: Soe.Angular/TypeScript/Common/Models/ActorSearchPersonDTO.ts
import { SoeEntityType } from "../../Util/CommonEnumerations";
import { IActorSearchPersonDTO } from "../../Scripts/TypeLite.Net4";
import { CalendarUtility } from "../../Util/CalendarUtility";


export class ActorSearchPersonDTO implements IActorSearchPersonDTO {
    consentDate: Date;
    entityType: SoeEntityType;
    entityTypeName: string;
    hasConsent: boolean;
    hasConsentString: string;
    isPrivatePerson: boolean;
    name: string;
    number: string;
    orgNr: string;
    recordId: number;

    public fixDates() {
        this.consentDate = CalendarUtility.convertToDate(this.consentDate);
    }
}

